SCons/Tool/dmd.py

```diff
@@ -136,7 +136,7 @@ def generate(env):
     env['_SHDLIBSONAME'] = '${DShLibSonameGenerator(__env__,TARGET)}'
     # NOTE: this is a quick hack, the soname will only work if there is
     # c/c++ linker loaded which provides callback for the ShLibSonameGenerator
-    env['DShLibSonameGenerator'] = SCons.Tool.ShLibSonameGenerator
+    env['DShLibSonameGenerator'] = env.get("SONAME_GENERATOR", SCons.Tool._LibSonameGenerator)('ShLib')
```

Comment:
This means you can only alter this behavior by setting SONAME_GENERATOR when the tool is initialized?
Likely that's not the desired functionality.
You should be able to do this when calling ```env.SharedLibrary(target, sources, SHNAME_GENERATOR=special_one)```